lib/log.c

```diff
@@ -202,42 +314,104 @@ static void vzlog_file(struct zlog *zl, struct timestamp_control *tsctl,
 	fflush(fp);
 }
 
+/* Search a buf for the filter strings */
+static int search_buf(const char *buf)
+{
+	char *found = NULL;
+
+	for (int i = 0; i < zlog_filter_count; i++) {
+		found = strstr(buf, zlog_filters[i]);
+		if (found != NULL)
+			return 0;
+	}
+
+	return -1;
+}
+
+/* Filter out a log */
+static int vzlog_filter(struct zlog *zl, struct timestamp_control *tsctl,
+			const char *proto_str, int priority, const char *format,
+			va_list args)
+{
+	int len = 0;
+	int ret = 0;
+	char buf[1024] = "";
+	va_list ac;
+
+	ret = time_print_buf(buf, len, sizeof(buf), tsctl);
+
+	len += ret;
+	if ((ret < 0) || ((size_t)len >= sizeof(buf)))
+		return search_buf(buf);
+
+	if (zl && zl->record_priority)
+		ret = snprintf(buf + len, sizeof(buf) - len,
+			       "%s: %s: ", zlog_priority[priority], proto_str);
+	else
+		ret = snprintf(buf + len, sizeof(buf) - len, "%s: ", proto_str);
+
+	len += ret;
+	if ((ret < 0) || ((size_t)len >= sizeof(buf)))
+		return search_buf(buf);
+
+	va_copy(ac, args);
+	ret = vsnprintf(buf + len, sizeof(buf) - len, format, ac);
+	va_end(ac);
+
+	len += ret;
+	if ((ret < 0) || ((size_t)len >= sizeof(buf)))
```

Comment:
This seems like a pointless check since we `return search_buf(buf)` regardless right?

Reply:
yep